internal/validation/mock_metric_validation.go

```diff
@@ -17,22 +17,35 @@ package validation
 import (
 	"fmt"
 	"reflect"
+	"regexp"
+	"strings"
 	"time"
 
 	"github.com/golang/protobuf/ptypes"
 
+	lbl "google.golang.org/genproto/googleapis/api/label"
 	"google.golang.org/genproto/googleapis/api/metric"
 	"google.golang.org/genproto/googleapis/monitoring/v3"
 	"google.golang.org/genproto/googleapis/rpc/errdetails"
 )
 
+// Time series constraints can be found at https://cloud.google.com/monitoring/quotas#custom_metrics_quotas.
 const (
-	// Time series constraints can be found at https://cloud.google.com/monitoring/quotas#custom_metrics_quotas.
+	maxLabelKeyLength            = 100
+	maxMetricTypeLength          = 200
+	maxNumberOfLabels            = 10
 	maxTimeSeriesPerRequest      = 200
 	maxTimeSeriesLabelKeyBytes   = 100
 	maxTimeSeriesLabelValueBytes = 1024
 )
 
+var (
+	labelKeyRegex           = regexp.MustCompile("[a-z][a-zA0-9_-]*")
+	serviceNameRegex        = regexp.MustCompile(`^([a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62}){1}(\.[a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62})*[\._]?$`)
```

Comment:
Was this regex sourced from somewhere? Can you add a link in a comment?

Reply:
Done